oraclizeAPI_0.4.sol

```diff
@@ -871,35 +871,29 @@ contract usingOraclize {
         return 0;
     }
 
-    function matchBytes32Prefix(bytes32 content, bytes prefix) internal returns (bool){
+    function matchBytes32Prefix(bytes32 content, bytes prefix, uint n_random_bytes) internal returns (bool){
         bool match_ = true;
-
-        for (var i=0; i<prefix.length; i++){
+        
+        for (var i=0; i< n_random_bytes; i++) {
```

Comment:
if n_random_bytes exceeds 256, this loop will overflow, consider using uint256 over var, if there's no other overflow or length checks on n_random_bytes